src/app/files.js

```diff
@@ -107,6 +107,47 @@ function Files (storage) {
     return files
   }
 
+  //
+  // Tree model for files
+  // {
+  //   'a': { }, // empty directory 'a'
+  //   'b': {
+  //     'c': {}, // empty directory 'b/c'
+  //     'd': { '/readonly': true, '/content': 'Hello World' } // files 'b/c/d'
+  //     'e': { '/readonly': false, '/path': 'b/c/d' } // symlink to 'b/c/d'
```

Comment:
Oh I think it might be confusing, because I still intend to use the underlying `rename`, `remove`, etc. with full path. After every change, `listAsTree` should be called by consumer to get an up to date view.

Now we could optimise this to have an internal representation in `Files` to avoid recalculation, if needed.

The consumer should have some means to compare two trees, but that is outside of files.